adapters/p2p2consensus/transaction.go

```diff
@@ -1,66 +1,56 @@
 package p2p2consensus
 
 import (
-	"errors"
 	"fmt"
 
 	"github.com/NethermindEth/juno/adapters/p2p2core"
 	consensus "github.com/NethermindEth/juno/consensus/types"
+	"github.com/NethermindEth/juno/core"
 	"github.com/NethermindEth/juno/core/felt"
-	"github.com/starknet-io/starknet-p2pspecs/p2p/proto/common"
+	"github.com/NethermindEth/juno/utils"
 	p2pconsensus "github.com/starknet-io/starknet-p2pspecs/p2p/proto/consensus/consensus"
 )
 
-func AdaptTransaction(t *p2pconsensus.ConsensusTransaction) (consensus.Transaction, error) {
-	if t == nil {
-		return consensus.Transaction{}, errors.New("transaction is nil")
-	}
+func AdaptTransaction(t *p2pconsensus.ConsensusTransaction, network *utils.Network) (consensus.Transaction, error) {
 	if err := validateConsensusTransaction(t); err != nil {
 		return consensus.Transaction{}, err
 	}
```

Comment:
I realized that this might not be ideal, but should this validate call here makes it confusing to me. 

For example, is this the right place to call validate, when this validate the only thing it does is to make sure all the respective methods of p2pConsensus are non nil.

Shouldn't this all checks be done upon receive of the message. It seems wrong to me that they are in adapters. Adapters should assume correctness of their inputs (not the validity of its data).

I cannot find where in the code this function is supposed to be called so I cannot propose a better architecture but opening the discussion here.